day6/adventcode2015_day6.py

```diff
@@ -0,0 +1,99 @@
+# this script runs faster with pypy if large arrays are used
+
+import re
+
+# getting a list of instructions and coordinates from external file
+with open('input.txt') as data:
+    instructions = data.readlines()
+
+
+def create_array(amount_of_rows: int, amount_of_columns: int, array_element=0) -> list:
+    """
+    This function returns an array that contains repeated, specified element.
+    :param amount_of_rows: how many rows should an array have
+    :param amount_of_columns: how many columns should an array have
+    :param array_element: what element should array contain, default is 0
+    """
+    the_array = [[array_element for _ in range(amount_of_columns)] for _ in range(amount_of_rows)]
+    return the_array if amount_of_rows != 0 and amount_of_columns != 0 else []
+
+
+def parse_numbers_from_string(input_string: str) -> list:
+    parsed_list = re.findall(r'\d+', input_string)
+    return [int(x) for x in parsed_list]
+
+
+def turn_on_light(input_number: int) -> int:
+    return 1 if input_number == 0 else input_number
+
+
+def turn_off_light(input_number: int) -> int:
+    return 0 if input_number == 1 else input_number
+
+
+def toggle_light(input_number: int) -> int:
+    return 1 if input_number == 0 else 0
+
+
+def check_how_many_lights_are_on(instruction_data: list, array_rows: int, array_columns: int) -> int:
+
+    lights_array = create_array(array_rows, array_columns)
+
+    for instruction in instruction_data:
+        y1, x1, y2, x2 = parse_numbers_from_string(instruction)
+        for row_index in range(x1, x2 + 1):
+            for column_index in range(y1, y2 + 1):
+                if 'turn on' in instruction:
+                    lights_array[row_index][column_index] = turn_on_light(lights_array[row_index][column_index])
+                elif 'turn off' in instruction:
+                    lights_array[row_index][column_index] = turn_off_light(lights_array[row_index][column_index])
+                else:
+                    lights_array[row_index][column_index] = toggle_light(lights_array[row_index][column_index])
```

Comment:
I've been looking for more efficient algorithms/data structures for this problem, and they do exist, but your code works fine and for now we can skip more complex solutions and get back to them somewhere in future. So I'm leaving the comment just FYI

Reply:
Also there is time and place for optimizations and for now pushing performance would be over-engineering ;)